src/Foundation/NSDictionary_2.cs

```diff
@@ -147,7 +147,28 @@ public TValue[] ObjectsForKeys (TKey[] keys, TValue marker)
 			return Runtime.GetNSObject<NSDictionary<TKey,TValue>> (_FromObjectsAndKeysInternal (objects.Handle, keys.Handle));
 		}
 
-		public static NSDictionary<TKey,TValue> FromObjectsAndKeys (TKey [] objects, TValue [] keys)
+		public static NSDictionary<TKey, TValue> FromObjectsAndKeys (TValue [] objects, TKey [] keys, nint count)
+		{
+			if (objects == null)
+				throw new ArgumentNullException (nameof (objects));
+			if (keys == null)
+				throw new ArgumentNullException (nameof (keys));
+			if (objects.Length != keys.Length)
+				throw new ArgumentException (nameof (objects) + " and " + nameof (keys) + " arrays have different sizes");
+			if (count < 1 || objects.Length < count || keys.Length < count)
```

Comment:
You already know that `objects.Length == keys.Length` so only one check against count is needed.